api/v1alpha1/application_webhook.go

```diff
@@ -55,6 +56,10 @@ var _ webhook.Validator = &Application{}
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
 func (r *Application) ValidateCreate() error {
 
+	// We use the DNS-1035 format for application names, so ensure it conforms to that specification
+	if len(validation.IsDNS1035Label(r.Name)) != 0 {
```

Comment:
Do we allow name updates?  I assume not but just checking if we need to validate on update

Reply:
Good question. Kubernetes doesn't allow metadata.name to be changed, so nothing we need to do here.